### PR TITLE
Rework code for portability (Linux, macOS, Windows)

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

**Comment by stweil** [comment]
One of the commits also removes the `-l deu`option, so Tesseract now takes the default language `eng` instead of `deu`. It would be even better to use `Latin` because that supports all languages which are based on Latin characters. Perhaps a user setting `extra parameters for tesseract` would help.

**Comment by luerhard** [comment]
That looks very nice ! And i totally agree that a parameter to set the language would help very much.

One thing that I experienced in addition: Currently all Files that are created during the process are kept. That includes the png-Files, the pdf. For my testfile that results in 35,8 MB for a 274 KB file. Assuming that scaling: If a perform OCR on all files of my 4GB zotero library (which I would want to search for stuff), it will grow to 520 GB. I would suggest to get rid of all the png files and to set an option if the *.ocr.pdf should get created at all (which takes around 15 MB of the 35.8).

**Comment by stweil** [comment]
@zuphilip, I added a commit to restore the indentation with tab characters in the JavaScript code.

**Comment by akavalar** [comment]
I'll chime in here too, because I have the same feedback as @luerhard - I'd love to see an option that gets rid of all extraneous stuff (image, hocr, text, etc. files) and keeps only the OCR-ed PDF file. Other than that this is awesome! :)

**Comment by stweil** [comment]
This is a rather young project, so it still has time to grow. :-)

File: chrome/content/zoteroocr.js [hunk 1]
```diff
@@ -1,20 +1,66 @@
...
+
+		// Look for the tesseract executable in the settings and at commonly used locations.
+		// If it is found, the settings are updated.
+		// Otherwise abort with an alert.
+		let ocrEngine = Zotero.Prefs.get("zoteroocr.ocrPath");
+		let found = false;
+		if (ocrEngine) {
+			found = yield OS.File.exists(ocrEngine);
+		} else {
+			let path = ["/usr/local/bin/", "/usr/bin/", "C:\\Program Files\\Tesseract-OCR\\", ""];
```
[the remark this answers is not in the document]
Comment (stweil) [reply chrome/content/zoteroocr.js]: The empty string is expanded to `tesseract` (or `tesseract.exe` on Windows) and would work if that executable is found in the current working directory.

I also considered adding the path which you suggested, but don't think it is relevant in times of Windows 10 and personal computers with at least 4 GiB. The current installers for Tesseract on Windows support 64 bit installs and don't use that path. Technically the addition is easy and could be done any time with a followup commit.

**Comment by zuphilip** [comment]
Thank you very much! This is merged now :rocket:

**Pull request merged**